scripts/py_matter_idl/matter_idl/generators/kotlin/MatterClusters.jinja

```diff
@@ -39,6 +39,71 @@
   {%- endif -%}
 {%- endmacro -%}
 
+{%- macro encode_tlv(encodable, tag, name, depth) %}
+  {%- if encodable.is_nullable and not encodable.is_optional -%}
+    {{name}}?.let {
+      {{encode_tlv(encodable.without_nullable(), tag, name, depth + 1)}}
+    }     
+  {%- elif encodable.is_optional and not encodable.is_nullable -%}
+    {{name}}?.let {
+      {{encode_tlv(encodable.without_optional(), tag, name, depth + 1)}}
+    }
+  {%- elif encodable.is_optional and encodable.is_nullable -%}
```

Comment:
nit: i wonder if you flip these if you can get rid of the "not", because if you do the "is_optional && is_nullable" first, then you can follow up with "is_optional" (and know it is not nullable because it is not both) and "is_nullable" separately.

We also seem to control the without, so maybe this code could be simplified using some `without_optional_or_nullable` method instead of separating without_nullable and without_optional.